apis/r/tests/testthat/test-SOMADataFrame.R

```diff
@@ -232,3 +232,76 @@ test_that("soma_joinid validations", {
     "soma_joinid field must be of type Arrow int64"
   )
 })
+
+test_that("platform_config is respected", {
```

Comment:
This is for `SOMADataFrame`.

I'll next need to copypasta this for `SOMASparseNDArray` and `SOMADenseNDArray` but I wouldn't mind some feedback first before I do that.